### ISSUE TITLE
kettle is down again

### ISSUE THREAD

**Issue by BenTheElder** [kubernetes/test-infra#24946]
**What happened**:

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/metrics-kettle/1479915842807271424

> ERROR: table k8s-gubernator:build.all is 6.8 hours old. Max allowed: 6 hours.
> ERROR: table k8s-gubernator:build.week is 6.9 hours old. Max allowed: 6 hours.
> ERROR: table k8s-gubernator:build.day is 6.9 hours old. Max allowed: 6 hours.

**What you expected to happen**:

kettle should run without error and process current results

**How to reproduce it (as minimally and precisely as possible)**:

https://prow.k8s.io/job-history/gs/kubernetes-jenkins/logs/metrics-kettle ?

**Please provide links to example occurrences, if any**:

see above

**Anything else we need to know?**:

https://github.com/kubernetes/test-infra/issues/23135
https://kubernetes.slack.com/archives/C09QZ4DQB/p1642705919011300

**Comment by BenTheElder** [comment]
This appears to be the most recent log for the kettle pod in google.com k8s-gubernator project g8r cluster:

> Info 2022-01-08 08:46:41.721 PST inserting build: gs://istio-circleci/test/439703

So it wedged inserting istio results again?

We should probably stop collecting results for external projects, this is what we hit last time as well https://github.com/kubernetes/test-infra/issues/23135#issuecomment-893015099

**Comment by BenTheElder** [comment]
there is also a patch here that never merged because it fails tests https://github.com/kubernetes/test-infra/pull/23460

**Comment by BenTheElder** [comment]
on the off chance this wedging is a fluke, I went ahead and deleted the pod to restart it (it's a deployment).

**Comment by BenTheElder** [comment]
It has been inserting builds happily, but it will be a long time before it is back up to date, even if nothing goes wrong. I intended to work on this problem, but other things came up, and I can't spare much time for this.

I *suspect* this is unnecessarily serial.

**Comment by BenTheElder** [comment]
... seems building kettle images is also broken https://github.com/kubernetes/test-infra/pull/24948#issuecomment-1018035026

**Comment by BenTheElder** [comment]
> b'Error while reading data, error message: JSON parsing error in row starting at position 681087456: Parser terminated before end of string'

> b'Error while reading data, error message: JSON parsing error in row starting at position 1658689589: Parser terminated before end of string'

> ERROR:root:error on gs://kubernetes-github-redhat/logs/ci-kubernetes-conformance-node-e2e-rhel/17731

We're removing this random out-of-project buckets from kettle (seems to be a legacy thing)  in https://github.com/kubernetes/test-infra/pull/24948, so we can try again once we can deploy that, currently pending #24954 merging tentatively.

**Comment by BenTheElder** [comment]
updated image is pushed with the new buckets.yaml, I am deploying to staging.

**Comment by BenTheElder** [comment]
Deploying didn't work because of different git short tag, also it requires the commit being deployed to be checked out even though we have automated image pushes, which is a bit clunky. I opted instead to override it with:

`make -C kettle update-staging TAG=v20220121-d744e91`

but we should probably switch to automated deployment and recording the desired tag in git instead.

**Comment by BenTheElder** [comment]
staging looks fine, updating prod `make -C kettle update TAG=v20220121-d744e91`

**Comment by BenTheElder** [comment]
So far it looks like we've inserted `gs://kubernetes/pr-logs/*` successfully without issue and moved on to `gs://kubernetes/logs/*`. I think we'll complete a LOT faster without all the misc non-kubernetes-project buckets after https://github.com/kubernetes/test-infra/pull/24948 and getting the new image deployed with this change.

**Comment by BenTheElder** [comment]
should be good again, even the manual poke yesterday got us temporarily unblocked on the rest of the pipeline (though kettle deployment got stuck again), but I think now it should be actually unstuck. it looks like we finished inserting `gs://kubernetes-jenkins/logs/*`.

https://prow.k8s.io/job-history/gs/kubernetes-jenkins/logs/metrics-kettle

**Comment by BenTheElder** [comment]
There are kettle logs as recent as `2022-01-22 23:12:53.628 PST`, we are still inserting builds happily and I see no errors / warnings.

We are a bit tight on disk space (843GB / 1TB) and should probably consider pruning the contents (remove logs from the removed buckets by prefix?), but it's working again and KETTLE is low priority at the moment ... We could always just increase the PD size yet again.
